FrenchQuizApp/Verbs/Views/AddVerbViewController.swift

```
@@ -0,0 +1,265 @@
+    }
+    
+    func displayEmptyVerbAlert () {
+        let alert = UIAlertController(title: "Enter Verb", message: "Please enter a Verb for both fields", preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Okay", style: .default, handler: nil))
+        self.present(alert, animated: true)
+    }
+    
+    func noVerbFoundAlert () {
+        let alert = UIAlertController(title: "Error", message: "Couldn't find verb \(String(describing: frenchTextField.text))  Please try again", preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Okay", style: .default, handler: nil))
+        self.present(alert, animated: true)
+    }
+    
+    func verbFoundAlert (_ dataString: NSString) {
+        let alert = UIAlertController(title: "Verb Found", message: "Would you like to save \(String(describing: frenchTextField.text))?", preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Okay", style: .default, handler: { action in
+            self.parseVerbConjugationFromDataString(dataString)
+        }))
+        self.present(alert, animated: true)
+    }
+    
+}
+
+extension AddVerbViewController: UITableViewDelegate, UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        guard  let sections = dataResultsController.sections else {
+            return 0
+        }
+        
+        return sections.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard let sectionInfo = dataResultsController.sections?[section] else {
+            fatalError("No sections in fetchedResultsController")
+        }
+        
+        return sectionInfo.numberOfObjects
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: constants.tableViewCellIdentifier, for: indexPath) as! VerbCell
```

Next time, try out:
```guard let cell = tableView.dequeueReusableCell(withIdentifier: constants.tableViewCellIdentifier, for: indexPath) as? VerbCell else {
    fatalError()
}```

it allows you to fail fast and works as a reminder, in case you added a new table cell type elsewhere but haven't implemented it yet.